Rails Blog App: Add forms

kkolade opened the pull request:
# In this PR, I:
- Create a method called current_user in the ApplicationController to make current user data available to all controllers.
- Create forms to perform the following functions:
   - Creates a Post on behalf of the current_user (using the method I created in the ApplicationController).
   - Create a comment on behalf of the current_user (using the method created in the ApplicationController).
- Allow Users to add likes to Posts.
- Add minimal styling to the forms and buttons created

kkolade committed:
Add currentuser method to Application controller

kkolade committed:
Modify posts resources entry in route.rb to handle all RESTful routes

kkolade committed:
Add "new" action to PostController

kkolade committed:
Add form to view template to create new post

kkolade committed:
Add "create" action to PostController with basic code necessary to submit a new post

kkolade committed:
Update Post#create to add success/error message and to permit only title and text submission

kkolade committed:
Add link to new post form to post#show controller

kkolade committed:
Refactor the form to place it in partial located in veiws/share

kkolade committed:
Add link to new post form on the user post page

kkolade committed:
Refactor routes.rb to add route for comments and likes resources

kkolade committed:
Add scaffold for comments controller

kkolade committed:
Add "new" and "create" def to the Comments controller

kkolade committed:
Implement Adding comments

kkolade committed:
Add likes controller through rails generate

kkolade committed:
Add code to likes#create

kkolade committed:
Add some more styling

kkolade committed:
Implement Like buttom

kkolade committed:
Fix linter errors

kkolade committed:
refactor models & controllers to correct typos

kkolade committed:
Run migration

kkolade committed:
Refactor code to display for user without posts

kkolade committed:
Refactor typo in comment model

kkolade committed:
Add codes to implement likes function

kkolade committed:
Add more styles and fix linter errors

kkolade committed:
Refactor JS to fade out alerts

kkolade committed:
Fix linter errors

Felix45 commented on app/views/comments/new.html.erb:
```diff
@@ -0,0 +1 @@
+<!-- Empty template, nothing to show -->
```
- [ Required ] I think it would be better if you follow the project requirements for this milestone, according to the project requirements the application should have a form in the `views/comments/new.html.erb` file which accepts input for creating `comments`. This means you should make sure that as much as a user can add a comment from the `posts show` page, when a user navigates to `GET /posts/:post_id/comments/new` the user should be able to add a comment to a post through a form also. Kindly implement this.

*[Hint] You can use the form on the posts show page*

Felix45 reviewed (changes requested):
Hi @kkolade, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://camo.githubusercontent.com/59fa9d95c18dbd5a8e47490a9e4b48393f569685410f3c78d4d83a9764fa1392/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f74385950663341434b656f754c7a454976722f67697068792e676966" width="350px" />

## Highlights :heavy_check_mark: 
- [x] Blog forms added correctly :white_check_mark: 
- [x] Application uses strong parameters in validating form input :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:   

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

kkolade committed:
Refactor code to implement issue raised by CR

Move comments form into a partial
Render the partial in show.html.erb
Render the partial form in new.html.erb
Refactor comments#new to add user and post id to params hash

kkolade committed:
Fix linter errors

Zuhaib042 reviewed (approved):
Hi @kkolade 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
![](https://media.giphy.com/media/QJsP0cTAJhg7O47ub2/giphy.gif)
## Highlights
✅ Implemented suggested changes
✅ Comments are getting submitted
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
